database/mongodb/mongodb.go

```diff
@@ -98,12 +98,16 @@ func (m *Mongo) Open(dsn string) (database.Driver, error) {
 
 func (m *Mongo) SetVersion(version int, dirty bool) error {
 	migrationsCollection := m.db.Collection(m.config.MigrationsCollection)
-	if err := migrationsCollection.Drop(context.TODO()); err != nil {
-		return &database.Error{OrigErr: err, Err: "drop migrations collection failed"}
-	}
-	_, err := migrationsCollection.InsertOne(context.TODO(), bson.M{"version": version, "dirty": dirty})
-	if err != nil {
-		return &database.Error{OrigErr: err, Err: "save version failed"}
+	filt := bson.D{{"version", bson.D{{"$exists", true}}}}
+	upd := bson.D{{"$set", bson.D{
+		{"version", int32(version)},
+		{"dirty", dirty},
+	}}}
+	var tr = true
+	if res := migrationsCollection.FindOneAndUpdate(context.TODO(), filt, upd, &options.FindOneAndUpdateOptions{Upsert: &tr}); res.Err() != nil {
```

Comment:
You'll probably run into fewer races with `FindOneAndUpdate()` + `Upsert: true` than `Drop()` + `InsertOne()`, but this change won't actually make it safe to run migrations concurrently. e.g. it doesn't serialize/coordinate changes

To run safely concurrently, you'll need to use transactions. This isn't a blocker for merging this PR, but as is, the release notes would only state that a race condition is less likely to happen.